samples/boards/nrf/clock_skew/src/main.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <sys/timeutil.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/nrf_clock_control.h>
+#include <drivers/counter.h>
+#include <nrfx_clock.h>
+
+#define TIMER_NODE DT_NODELABEL(timer0)
```

Comment:
This is covered in the README: 

> This sample uses the API for correlating time sources to measure the skew between HFCLK (used for the CPU) and LFCLK (used for system time).

It's pretty specific to Nordic, because it's an important issue when `k_busywait()` uses a different clock than `k_sleep()`.

Really all you need is "counter"s from two different sources.

Reply:
Okay, so then
-  `nordic_nrf_clock` is the chosen configured SoC system clock (through clock_control API?) == HFCLK (external oscilator) = used for CPU
- `timer0` is chosen through `CONFIG_COUNTER_TIMER0` (which seems to be part of a counter API driver, `counter_nrfx_timer`) == LFCLK = used for system timer

My guess is that with this set up, this sample also wouldn't be compatible with NRF's power management, right (if the HFCLK is shut off in the chosen sleep mode, at least)? Or is there a NRF-specific reason this wouldn't be an issue?

Reply:
In Nordic neither HFCLK nor LFCLK are ever turned off.  They can theoretically change their source, which would make them monotonic and continuous but not *linear*, which is a pain, but the clock API isn't really set up for doing that so it's unlikely to happen.

Reply:
Ah alright. But it might still be problematic for other SoCs/Boards (like the STM32L1x SoCs). IMO it should be stated somewhere really clearly that this API is only compatible with power management if requirements xyz are met. 

From the top of my head, something along the lines of:
- if API is to be used throughout application: system timer source & external time source both aren't discontinuous (don't shut off during sleep)
- if either system timer source or external time source are discontinuous (due to shutting off during sleep), the API can be used only at boot (so before either sources has shut off) - (+ potentially the suggestions we discussed above, with the calibration skew calculation)

I have a feeling this doesn't cover everything, but just as an idea.